### ISSUE TITLE
Design heat losses

### ISSUE THREAD

**Issue by rubenbaetens** [open-ideas/IDEAS#241]
We'll add the design heat loss computation as parameter in the building components, and propagate it to the interfaces together with the ventilation system to (semi-)automate the computation of the overall design heat loss per zone. As such, we can more easily do a parametric run for buildings with a heating system, even at neighborhood scale.

@GlennReynders has a first implementation of this.

**Comment by cprotopa** [comment]
Does anyone know if it's possible to give the value of a **variable** to a **parameter**? @Mathadon?  Like for example the initial value or so? It would be useful to connect the Qnominal through the PropsBus, but then make it a parameter to be used in the heating systems.

**Comment by rubenbaetens** [comment]
@cprotopa i found it, the key is to use ```(fixed=false)```:
```
model Unnamed
  parameter Real a(fixed=false);
  Real b = time;
initial equation
  a = b+10;
end Unnamed;
```

**Comment by cprotopa** [comment]
@damienpicard @Mathadon Is it a problem for you if I remove the `parameter QNom` from the `Partial_IdealHeating` and `Partial_HydraulicHeating` and put it directly in the `Interfaces.BaseClasses.HeatingSystem` ? By doing this I could then give it a default `QNom=building.QNom` in the `Building` interface. If you agree.

**Comment by damienpicard** [comment]
@cprotopa I wouldn't like to put to much in the `Interfaces.BaseClasses.HeatingSystem` nor in `Interface.Building`, because I use them too and `QNom` makes no sense to me, as I'm using multiple systems.

If it is important for you and all your heatingSystems have this parameter, you could make an extension of `Building` where you add the parameter `QNom` and add it as an argument to the heatingSystem.

**Comment by Mathadon** [comment]
I agree to Damien’s suggestion :)

**Comment by rubenbaetens** [comment]
@damienpicard @Mathadon actually, i rather have a more content-full answer than *"I don't need it for my special case so i wouldn't do it"*. In that case there wouldn't be an electrical system in the interface either as none of you are doing `ID-EAS`. It's just the implementation of the design heat loss as in `EN 12831`, so not actually `Q_nominal` but `Q_design`, if that's no usefull information (even in let's say post-processing) ...

**Comment by damienpicard** [comment]
@cprotopa @rubenbaetens : what I would like to avoid it to increase the number of parameters you need to fill in, especially if their use are case specific. For example, the parameter VZones was in heating system (without default value), while lots of heating systems do not need such a parameter.

Adding parameter to `Structure` which calculates the design losses of the structure is a different matter. That could definitely be useful and we not require any input of the users.

**Comment by rubenbaetens** [comment]
@damienpicard Well, that's the point of it; The interfaces' baseclasses `HeatingSystem`, `VentilationSystem` etcetera are made to be part of the `Building`-interface. If there is e.g. a `Q_design` parameter in the interfaces' baseclass `HeatingSystem`, it's explicitely there nót to be filled in by the user, but by default to be propagated in the `Buildings`-interface as `replaceable heatingSystem(Q_design=structure.Q_design+ventilationSystem.Q_design)`.

**Comment by Mathadon** [comment]
I have to agree with Damien that it does not make sense to specify a nominal power for a heating system. What would you use this new parameter for in the heating system? If it's for dimensioning a boiler or for post-processing then I suggest you extend the HeatingSystem such as suggested by @damienpicard.

It's good that this parameter would have a default value but that still leaves you with a parameter in the HeatingSystem that is not used by default.

**Comment by rubenbaetens** [comment]
OK, so i conclude most of you don't use `HeatingSystem`-baseclass by default as part of the `Buildings`-interface. In that case we could define a `Boolean IDEAS = true;` in the `Buildings`-interface and define those 'undesired' parameters as `annotation(Dialog(enable=IDEAS));` so that they only exist if properly propagated in the `Buildings`-interface (?).

**Comment by rubenbaetens** [comment]
Closed in #277.
